playwright/tests/utils.py

```diff
@@ -0,0 +1,111 @@
+"""
+Useful functions for testing.
+"""
+
+from datetime import datetime
+from pathlib import Path
+
+
+def get_hugo_species_dir() -> Path:
+    """
+    Get the Hugo species directory.
+    """
+    return Path(__file__).parent.parent.parent / "hugo/content/species"
+
+
+def get_list_of_species() -> list[str]:
+    """
+    Search the Hugo content directory to get all the species on the website.
+    This helper function is useful for testing each species page in the same way.
+
+    The function also validates each page is not in draft mode (wont show up on website yet)
+    """
+    HUGO_SPECIES_DIR = get_hugo_species_dir()
+
+    species = []
+    for folder in HUGO_SPECIES_DIR.iterdir():
+        if not folder.is_dir():
+            continue
+
+        index_file = folder / "_index.md"
+        if not index_file.exists():
+            continue
+
+        with index_file.open() as file_in:
+            content = file_in.read()
+            if "draft: true" not in content:
+                species.append(folder.name)
+
+    return species
+
+
+SPECIES_LIST = get_list_of_species()
+
+
+def all_intro_page_paths() -> list[str]:
```

Comment:
A general question: why did you prefer functions that return constant values rather than module-level constants ? For example, `get_hugo_species_dir()` rather than `HUGO_SPECIES_DIR = ...`

Reply:
Thanks for this, it never occurred to me that you could just import the constant into other files *facepalm*. I've done that now in commit 01c94dfccc410fbafdf51d849cb6d28f95fed5fa